tests/lib/rules/no-whitespace-before-property.js

```diff
@@ -507,6 +510,46 @@ ruleTester.run("no-whitespace-before-property", rule, {
             code: "foo ['bar ' + 1 + ' baz']",
             output: "foo['bar ' + 1 + ' baz']",
             errors: ["Unexpected whitespace before property 'bar ' + 1 + ' baz'."]
+        },
+        {
+            code: "5       .toExponential()",
+            output: "5 .toExponential()",
```

Comment:
Warning the patterns sounds good to me.
But in that case, I think the rule should not fix it automatically since there are variety candidates to fix.